crates/testenv/src/lib.rs

```diff
@@ -186,6 +186,22 @@ impl TestEnv {
         }
     }
 
+    /// This method waits for Electrum to see a transaction with given `txid`.
+    pub fn wait_until_electrum_sees_txid(&self, txid: Txid) -> anyhow::Result<()> {
+        let mut delay = Duration::from_millis(64);
+
+        loop {
+            if self.electrsd.client.transaction_get(&txid).is_ok() {
+                return Ok(());
+            }
```

Comment:
This will create a situation where we may loop forever.

Are we able to figure out if the error is a not-exist error? Ideally, the `rust_esplora` crate has something to figure this out easily. If not, we should upstream some changes later.

Also, would be good to add a timeout as an input. Just in case the test fails (the txid never shows up). We at least want to exit the test!

Reply:
The timeout thing can be added in a separate PR. Can also add timeouts to the other wait methods.

Reply:
The error returned is: `Err(Protocol(String("missing transaction")))`